src/factory/index.ts

```diff
@@ -0,0 +1,80 @@
+import { SQLTemplate } from '@src/types';
+import { createListOfSqlParams } from '@src/util';
+import { OperatorValue, Queryable, QueryOptions } from './types';
+import { where } from './where';
+
+const parseOptions = <T extends Queryable>(
```

Comment:
Parse options should be at the bottom of this document according to clean code practices